File: src/test/java/compiler/parser/comments/CommentParser.java
package compiler.parser.comments;

import compiler.block.Block;
import compiler.block.comments.CommentBlock;
import compiler.parser.Parser;
import compiler.tokenizer.Tokenizer;

public class CommentParser extends Parser<CommentBlock> {

    @Override
    public boolean shouldParse(String line) {
        return line.matches("//.*");
    }

    @Override
    public CommentBlock parse(Block superBlock, Tokenizer tokenizer) {
        return new CommentBlock(superBlock);
    }
}

File: src/main/scala/compiler/parser/comments/CommentParser.scala
package compiler.parser.comments

import compiler.block.Block
import compiler.block.comments.CommentBlock
import compiler.parser.Parser
import compiler.tokenizer.Tokenizer
import test_classes.block.Block
import test_classes.block.comments.CommentBlock
import test_classes.parser.Parser
import test_classes.tokenizer.Tokenizer

class CommentParser extends Parser[CommentBlock] {

  def shouldParse(line: String): Boolean = line.matches("//.*")

  def parse(superBlock: Block, tokenizer: Tokenizer): CommentBlock = new CommentBlock(superBlock)

}
